src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -833,33 +828,39 @@ public void shutdown() throws InterruptedException {
     @Getter
     @Data
     static class CurrentProcessingFileInformation {
-        @JsonProperty(PERSISTED_CURRENT_PROCESSING_FILE_NAME)
-        private String fileName;
         @JsonProperty(PERSISTED_CURRENT_PROCESSING_FILE_START_POSITION)
         private long startPosition;
         @JsonProperty(PERSISTED_CURRENT_PROCESSING_FILE_LAST_MODIFIED_TIME)
         private long lastModifiedTime;
+        @JsonProperty(PERSISTED_CURRENT_PROCESSING_FILE_HASH)
+        private String fileHash;
 
         public Map<String, Object> convertToMapOfObjects() {
             Map<String, Object> currentProcessingFileInformationMap = new HashMap<>();
-            currentProcessingFileInformationMap.put(PERSISTED_CURRENT_PROCESSING_FILE_NAME, fileName);
             currentProcessingFileInformationMap.put(PERSISTED_CURRENT_PROCESSING_FILE_START_POSITION, startPosition);
             currentProcessingFileInformationMap.put(PERSISTED_CURRENT_PROCESSING_FILE_LAST_MODIFIED_TIME,
                     lastModifiedTime);
+            currentProcessingFileInformationMap.put(PERSISTED_CURRENT_PROCESSING_FILE_HASH, fileHash);
             return currentProcessingFileInformationMap;
         }
 
         public void updateFromTopic(Topic topic) {
             switch (topic.getName()) {
-                case PERSISTED_CURRENT_PROCESSING_FILE_NAME:
-                    fileName = Coerce.toString(topic);
-                    break;
                 case PERSISTED_CURRENT_PROCESSING_FILE_START_POSITION:
                     startPosition = Coerce.toLong(topic);
                     break;
                 case PERSISTED_CURRENT_PROCESSING_FILE_LAST_MODIFIED_TIME:
                     lastModifiedTime = Coerce.toLong(topic);
                     break;
+                case PERSISTED_CURRENT_PROCESSING_FILE_HASH:
+                    // This handles the upgrade scenario. ALl stored fileNames are transferred to the fileHash.
+                    String savedString = Coerce.toString(topic);
```

Comment:
Instead of deleting the old key on this release we should keep writing to both and but only read from the hash. In base someone upgrades and then downgrades at least it would keep working

Reply:
Agree. I remove it first, and need to handle it in a separate PR.